### ISSUE TITLE
Improve VirtualMap detach

### ISSUE THREAD

**Issue by akugal** [hashgraph/hedera-services#24250]
`VirtualMap.detach() `
- Must be used only when the pipeline is paused, so pausing the pipeline has to be part of this method
- The detach method can be removed from the `VirtualRoot` interface
- Missing some tests
